packages/mongo/oplog_v2_converter.js

```diff
@@ -0,0 +1,113 @@
+// we are mapping the new oplog format on mongo 5
+// to what we know better, $set and $unset format
+// new oplog format ex:
+// {
+//  '$v': 2,
+//  diff: { u: { key1: 2022-01-06T18:23:16.131Z, key2: [ObjectID] } }
+// }
+
+/*
+the structure of an entry is:
+
+
+-> entry: i, u, d + sFields.
+-> sFields: i, u, d + sFields
+-> sFields: arrayOperator -> { a: true, u0: 2 }
+-> i,u,d: { key: value }
+-> value: {key: value}
+
+i and u are both $set
+d is $unset
+on mongo 4
+ */
+
+const isArrayOperator = possibleArrayOperator => {
+  if (!Object.keys(possibleArrayOperator).length) return false;
+
+  let isArrayOperator = true;
+  Object.keys(possibleArrayOperator).forEach(key => {
+    if (key !== 'a' && !key.match(/u\d+/)) {
+      isArrayOperator = false;
+    }
+  });
+  return isArrayOperator;
+};
+const nestedOplogEntryParsers = (
+  { i = {}, u = {}, d = {}, ...sFields },
+  prefixKey = ''
+) => {
+  const sFieldsOperators = [];
+  Object.entries(sFields).forEach(([key, value]) => {
+    if (isArrayOperator(value || {})) {
+      const { a, ...uPosition } = value;
+      const [positionKey, newArrayIndexValue] = Object.entries(uPosition)[0];
+      if (uPosition)
```

Comment:
it's missing {

And when `uPosition` will not be truth value?

Reply:
never. I have added a throw in this case.